README.rst

```diff
@@ -686,13 +686,17 @@ variable ``FUN``:
         is given, then all the ``bin`` passed will be prepended with this
         ``path``. ``bin``\ s can be passed as multiple arguments or as one
         list.
-``mapfiles(src, dst, file[, ...][, append_suffix=True])``
+``mapfiles(src, dst, [file[, ...]][, append_suffix=True])``
         A very simple function that just returns a list with
         ``{src}/{file}={dst}/{file}{VAR.suffix}`` for each ``file`` passed.
-        ``file``\ s can be passed as multiple arguments or as one list. A named
-        argument ``append_suffix`` can be passed at the end to control whether
-        ``VAR.suffix`` is appended to each destination file. ``append_suffix``
-        defaults to ``True`` if not given.
+        ``file``\ s can be passed as multiple arguments or as one list. If no
+        ``file``\ s are passed, a simple ``{src}/={dst}/`` mapping is returned
+        which results in the directory structure under ``src`` (including all
+        contained files) being replicated as-is from the local filesystem into
+        ``dst`` within the package. A named argument ``append_suffix`` can be
+        passed at the end to control whether ``VAR.suffix`` is appended to each
+        destination file. ``append_suffix`` defaults to ``True`` if not given,
+        and is only applicable when at least one ``file`` is passed.
```

Comment:
Mmm, I don't like that this function have so many different behaviours now based on the combination of options you pass. And honestly using `FUN.mapfiles('src', 'dst')` vs using `"src/=dst/"` makes little to no sense (or am I missing something here?). If the suffix were somehow handled at least (but I don't even know then if it should be added to the `dst` directory, but in that case is the same as `"src/=dst"+VAR.suffix+"/"`, which doesn't seems particularly complicated either).